src/include/msvc.h

```diff
@@ -34,6 +34,47 @@
 #define WT_GCC_FUNC_ATTRIBUTE(x)
 #define WT_GCC_FUNC_DECL_ATTRIBUTE(x)
 
+#define WT_UNUSED_MSVC(var) (void)(var)
+
+#define WT_ATOMIC_RELAXED 0
+#define WT_ATOMIC_ACQUIRE 1
+#define WT_ATOMIC_RELEASE 2
+#define WT_ATOMIC_ACQ_REL 3
+#define WT_ATOMIC_SEQ_CST 4
+
+#define WT_C_MEMMODEL_ATOMIC_LOAD(var, ptr, memorder) \
+    do {                                              \
+        var = *(ptr);                                 \
+        switch (memorder) {                           \
+        case WT_ATOMIC_RELAXED:                       \
+        case WT_ATOMIC_RELEASE:                       \
+            break;                                    \
+        case WT_ATOMIC_ACQUIRE:                       \
+        case WT_ATOMIC_ACQ_REL:                       \
+            WT_BARRIER();                             \
+            break;                                    \
+        case WT_ATOMIC_SEQ_CST:                       \
+            WT_FULL_BARRIER();                        \
```

Comment:
I'm not sure whether here we need a full barrier or just a read barrier is enough to get the same guarantee of WT_ATOMIC_SEQ_CST.

Reply:
Remember that you NEVER need read or write barriers on x86. The lfence and sfence instructions do exist, but aren't really for user-space concurrency. If you use the correct instructions for writes then you don't need any fencing for loads on x86, even at seq_cst.

Reply:
ok, understood.